build-system/tasks/check-video-interface-list.js

```diff
@@ -0,0 +1,125 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const argv = require('minimist')(process.argv.slice(2));
+const globby = require('globby');
+const tempy = require('tempy');
+const {bold, blue, yellow} = require('kleur/colors');
+const {getStdout} = require('../common/process');
+const {log, logWithoutTimestamp} = require('../common/logging');
+const {readFile, writeFile} = require('fs-extra');
+
+const filepath = 'spec/amp-video-interface.md';
+
+const excludeGeneric = ['amp-video', 'amp-video-iframe'];
+
+const grepJsContent = '"@implements {.*VideoInterface}"';
+
+const grepJsFiles = 'extensions/**/*.js';
+
+// File should have a section containing only entries.
+// - [amp-whatever](https://amp.dev/documentation/components/amp-whatever)
+const entry = (name) =>
+  `-   [${name}](https://amp.dev/documentation/components/${name})`;
+
+const findVideoInterfaceExtensions = () =>
+  getStdout(['grep -lr', grepJsContent, ...globby.sync(grepJsFiles)].join(' '))
+    .trim()
+    .split('\n')
+    .map((path) => path.substr('extensions/'.length).split('/').shift())
+    .filter((name) => !excludeGeneric.includes(name));
+
+function getSectionRange(content) {
+  const entryRegExp = new RegExp(
+    entry('$NAME')
+      .replace(/[\[\]\(\)]/g, (c) => `\\${c}`)
+      .replace(/\$NAME/g, '([a-z0-9-]+)') + '[\\n\\s]+',
+    'gim'
+  );
+
+  let match;
+  let sectionStart = -1;
+  let sectionEnd = 0;
+  while ((match = entryRegExp.exec(content)) !== null) {
+    const [line, name] = match;
+    const {index} = match;
+    if (excludeGeneric.includes(name)) {
+      continue;
+    }
+    if (sectionStart < 0) {
+      sectionStart = index;
+      sectionEnd = index;
+    }
+    if (sectionEnd < index) {
+      break;
+    }
+    sectionEnd += line.length;
+  }
+  return [sectionStart, sectionEnd];
+}
+
+function expected(content) {
+  const [sectionStart, sectionEnd] = getSectionRange(content);
+  return (
+    content.substr(0, sectionStart) +
+    findVideoInterfaceExtensions().map(entry).join('\n') +
+    '\n\n' +
+    content.substr(sectionEnd)
+  );
+}
+
+async function diffTentative(filepath, content) {
+  const temporary = await tempy.write(content);
+  return getStdout(
+    `git -c color.ui=always diff -U1 ${filepath} ${temporary}`
+  ).replace(new RegExp(temporary, 'g'), `/${filepath}`);
+}
+
+async function checkVideoInterfaceList() {
+  const content = (await readFile(filepath)).toString('utf-8');
```

Comment:
```suggestion
  const content = await readFile(filepath, 'utf-8');
```

See https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options